File: Exposure/Exposure.Api/Exposure.Api/Models/Picture.cs
using SqlSugar;

namespace Exposure.Api.Models;

[SugarTable("Picture")]
public class Picture
{
    [SugarColumn(IsPrimaryKey = true, IsIdentity = true)]
    public int Id { get; set; }

    /// <summary>
    ///     用户ID
    /// </summary>
    public int UserId { get; set; }

    /// <summary>
    ///     图片名称
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    ///     图片路径
    /// </summary>
    public string Path { get; set; }
    
    /// <summary>
    ///     缩略图
    /// </summary>
    public string Thumbnail { get; set; }

    /// <summary>
    ///     宽度
    /// </summary>
    public int Width { get; set; }

    /// <summary>
    ///     高度
    /// </summary>
    public int Height { get; set; }

    public int Type { get; set; }

    /// <summary>
    ///     曝光时间
    /// </summary>
    public int ExposureTime { get; set; }

    /// <summary>
    ///     曝光增益
    /// </summary>
    public int ExposureGain { get; set; }

    /// <summary>
    ///     白平衡
    /// </summary>
    public int BlackLevel { get; set; }

    /// <summary>
    ///     是否删除
    /// </summary>
    public bool IsDelete { get; set; }

    /// <summary>
    ///     创建时间
    /// </summary>
    public DateTime CreateTime { get; set; }

    /// <summary>
    ///     更新时间
    /// </summary>
    public DateTime UpdateTime { get; set; }

    /// <summary>
    ///     删除时间
    /// </summary>
    public DateTime DeleteTime { get; set; }
}

File: CSharp/Exposure/Exposure.App/src/renderer/src/api/picture.ts
import axios from 'axios'

// 图片
export interface Picture {
  id: number
  userId: number
  name: string
  path: string
  width: number
  height: number
  type: number
  thumbnail: string
  exposureTime: number
  exposureGain: number
  blackLevel: number
  isDelete: boolean
  createTime: string
  updateTime: string
  deleteTime: string
}

// 图片库
export interface PictureGallery {
  date: string
  light: Picture[]
  dark: Picture[]
  combine: Picture[]
}

// 图片参数
export interface PictureQueryParam {
  page: number
  size: number
  isDeleted: boolean
  name: string
  startTime: string | null
  endTime: string | null
}

// 图片分页结果
export interface PicturePageResult {
  total: number
  list: Picture[]
}

// 图片导出参数
export interface PictureExportParam {
  ids: number[]
  format: string
}

// 图片调整参数
export interface PictureAdjustParam {
  id: number
  brightness: number
  contrast: number
  invert: boolean
  code: number
}

// 图片更新参数
export interface PictureUpdateParam {
  id: number
  name: string
}

// 获取图片
export function getByPage(data: PictureQueryParam) {
  return axios.post<PicturePageResult>('/Picture/Page', data)
}

// 更新图片
export function updatePicture(data: PictureUpdateParam) {
  return axios.put('/Picture', data)
}

// 删除图片
export function deletePicture(data: number[]) {
  return axios.delete('/Picture', { data })
}

// 合并图片
export function combinePicture(data: number[]) {
  return axios.post<Picture>('/Picture/Combine', data)
}

// 导出图片
export function exportPicture(data: PictureExportParam) {
  return axios.post('/Picture/Export', data)
}

// 调整图片
export function adjustPicture(data: PictureAdjustParam) {
  return axios.post('/Picture/Adjust', data)
}
